torch/_inductor/fx_passes/post_grad.py

```diff
@@ -102,9 +102,11 @@ def post_grad_passes(gm: torch.fx.GraphModule, is_inference: bool):
 
     fake_tensor_updater.incremental_update()
 
-    # Keep this last, since it introduces mutation. Look at
+    # Keep these last, since they introduces mutation. Look at
     # ./fx_passes/README.md for a discussion of mutation invariants.
     reinplace_inplaceable_ops(gm.graph)
+    decompose_auto_functionalized(gm.graph)
```

Comment:
Is reinplacing mutable custom ops worth deliberating at all now (vs future work)?

If we eventually want the `reinplace_Inplaceable_ops` pass to handle reinplacing custom ops, then I guess we have two options: add special logic to to the reinplacing pass to understand `auto_functionalize` and try to reinplace it, or desugar into `clone` + `inplace_op` **before** the reinplacing pass, and try to get that pass to get rid of unnecessary clones.

Reply:
This PR is an improvement to the current state and does not make the future reinplacing work more difficult, so we should review it without worrying about the future reinplacing work.